models/animals.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+
+class Animal(ABC):
+    def __init__(self, name: str, weight: float):
+        self._name = name
+        self._weight = weight
```

Comment:
Why _protected though? It doesn't mean "protected" as in Java, it means "hey fellow coders and teammates, please don't touch this value, it's temporary or holds some internal data needed for calculations/API usage etc."